File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/4 - 1 and 2 December 2018/02_mountain_run.py
import math

record_in_seconds = float(input())
distance_meters = float(input())
time_seconds_for_one_meter = float(input())

georgi_time_clean = distance_meters * time_seconds_for_one_meter
georgi_time_minus_50_seconds = math.floor(distance_meters / 50) * 30
georgi_time = georgi_time_clean + georgi_time_minus_50_seconds

if georgi_time < record_in_seconds:
    print(f"Yes! The new record is {georgi_time:.2f} seconds.")
else:
    print(f"No! He was {abs(record_in_seconds - georgi_time):.2f} seconds slower.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/08. Online Exam - 28 and 29 march 2020/02. Mountain Run.js
function mountainRun(input) {
    // Mask
    let record = Number(input.shift());
    let distance = Number(input.shift());
    let timePerMeter = Number(input.shift());

    let time = (distance * timePerMeter) + (Math.floor(distance / 50) * 30);

    if (time < record) {
        console.log(`Yes! The new record is ${time.toFixed(2)} seconds.`);
    } else {
        console.log(`No! He was ${(time - record).toFixed(2)} seconds slower.`);
    }
}

mountainRun([10164, 1400, 25]);  // Should return:
// No! He was 25676.00 seconds slower.

mountainRun([5554.36, 1340, 3.23]);  // Should return:
// Yes! The new record is 5108.20 seconds.
